src/throttledStore.tsx

```diff
@@ -64,11 +216,58 @@ export const createThrottledStore = (
         notifySubscribers()
     }
 
-    return _.defaults(
-        {
-            subscribe,
-            flush
+    const dispatch: Dispatch<AnyAction> = action => {
+        resetPendingNotifications()
+        const dispatchResult = store.dispatch(action)
+        return dispatchResult
+    }
+
+    const result: PrivateThrottledStore = {
+        ...store,
+        subscribe,
+        dispatch,
+        flush,
+        broadcastNotify: onBroadcastNotify,
+        observableNotify: onObservableNotify,
+        resetPendingNotifications
+    }
+
+    resetPendingNotifications()
+    return result
+}
+
+export const createObservable = <TState, TSelector>(
+    shell: Shell,
+    uniqueName: string,
+    selectorFactory: (state: TState) => TSelector
+): PrivateObservableState<TState, TSelector> => {
+    const subscribersSlotKey: SlotKey<StateObserver<TSelector>> = {
+        name: uniqueName
+    }
+    const observersSlot = shell.declareSlot(subscribersSlotKey)
+    let cachedSelector: TSelector | undefined
+
+    const getOrCreateCachedSelector = (): TSelector => {
+        if (cachedSelector) {
+            return cachedSelector
+        }
+        const newSelector = selectorFactory(shell.getStore<TState>().getState())
+        cachedSelector = newSelector
+        return newSelector
+    }
+
+    return {
+        subscribe(fromShell, callback) {
+            observersSlot.contribute(fromShell, callback)
+            return () => {
+                observersSlot.discardBy(item => item.contribution === callback)
+            }
+        },
+        notify() {
+            cachedSelector = undefined
+            const newSelector = getOrCreateCachedSelector()
+            invokeSlotCallbacks(observersSlot, newSelector)
```

Comment:
I think the word `selector` is really confusing, does this mean `selected`?

Reply:
I'm open for suggestions :)
Technically `StateObservable<T>` can be of any `T`, but the pattern is `StateObservable<TSelectorAPI>` where `TSelectorAPI` is a public interface that contains some selector functions over the observable state. So the "selector" is an object that implements the `TSelectorAPI` interface.